### DISCUSSION TITLE
Mustache Templates and msg property naming convention

### DISCUSSION THREAD

**Discussion by xs539** [zachowj/node-red-contrib-home-assistant-websocket#1691] [Q&A] [answered]
Node-RED version: 4.0.5
node-red-contrib-home-assistant-websocket version: 0.74.2

My notify Action node stopped working: 
![image](https://github.com/user-attachments/assets/8a41a016-1c13-47ac-aa71-e623075aa95f)
![image](https://github.com/user-attachments/assets/75724b50-ef4b-486f-b5f0-2655d49cbca4)

It seems it is not able to parse camelCase properties, when I change the property to all lowercase (alerttargetpush) it does in fact work again.

What is the naming convention for message properties in Mustache Templates? 
The [docs](https://zachowj.github.io/node-red-contrib-home-assistant-websocket/guide/mustache-templates.html) show only lowercase examples but it does not state anything about uppercase.

Should I avoid using uppercase characters?

**Comment by zachowj** [comment]
This looks like a bug. The action string is converted to lowercase before any template variables are rendered.

https://github.com/zachowj/node-red-contrib-home-assistant-websocket/blob/43d3e091ea67eee51c84ea2c8c836b9dbb862e07/src/nodes/action/ActionController.ts#L48

**Comment by zachowj** [comment, marked as answer]
This has been fixed in the latest release.

**Reply by xs539** [reply]
Awesome, thank you!
